### PR TITLE
Adding REXTwoWayUniswapMarket

### PR DESCRIPTION
**Pull request by shreyaspapi**
- [x] Tests remaining

### CODE REVIEW

**Comment by shreyaspapi** [comment]
fixes #98, waiting for REXSuperSwap to complete.

**Comment by mikeghen** [comment]
@shreyaspapi, Im going to start to look at migrating to these uniswapv3 contracts, I'm working on getting #134 merged then I'll merge this in too, in the meantime, could you fix the merge conflict here?

**Comment by shreyaspapi** [comment]
@mikeghen  I have rebased it.

File: .gitignore [hunk 1]
```diff
@@ -12,4 +12,4 @@ coverage
 coverage.json
 typechain
 typechain-types
-
```
Comment (mikeghen) [inline .gitignore:15]: This can be removed

**Review by mikeghen — COMMENTED** [review]
Thanks @shreyaspapi, sorry this is taking me so long to get to review, I just cloned your repo to run the tests.

First, I think you need to upgrade the superfluid SDK, when I cloned a fresh repo and got the latest version of the SDK there were some errors, you'll have to make a few changes I highlight them in these image:
Screen Shot 2022-12-05 at 2 57 26 PM
Screen Shot 2022-12-05 at 2 57 20 PM

Also, after fixing these issues, I get some test errors that look like this, would you take a look at these? Im also going to look at them myself and see if I can find the source of the error, maybe something related to me using the latest version of the SDK.
```

1) REXTwoWayUniswapMarket
       #1 - new rexmarket with no streamers
         #1.2 before/afterAgreementCreated callbacks:
     AssertionError: Expected "0" to be equal 1000000000000000
      at Context.<anonymous> (test/REXUniswapTwoWayMarket.test.ts:444:18)
      at runMicrotasks (<anonymous>)
      at processTicksAndRejections (internal/process/task_queues.js:95:5)

  2) REXTwoWayUniswapMarket
       #1 - new rexmarket with no streamers
         #1.4 one-sided distribution:
     Error: Transaction reverted: function returned an unexpected amount of data
      at REXTwoWayUniswapMarket._swap (contracts/REXUniswapTwoWayMarket.sol:402)
      at REXTwoWayUniswapMarket.distribute (contracts/REXUniswapTwoWayMarket.sol:196)
      at runMicrotasks (<anonymous>)
      at processTicksAndRejections (internal/process/task_queues.js:95:5)
      at async HardhatNode._mineBlockWithPendingTxs (node_modules/hardhat/src/internal/hardhat-network/provider/node.ts:1649:23)
      at async HardhatNode.mineBlock (node_modules/hardhat/src/internal/hardhat-network/provider/node.ts:458:16)
      at async EthModule._sendTransactionAndReturnHash (node_modules/hardhat/src/internal/hardhat-network/provider/modules/eth.ts:1496:18)
      at async HardhatNetworkProvider.request (node_modules/hardhat/src/internal/hardhat-network/provider/provider.ts:117:18)
      at async TracerWrapper.request (node_modules/hardhat-tracer/src/wrapper.ts:36:16)

  3) REXTwoWayUniswapMarket
       #2 - existing market with streamers on both sides
         #2.1 before/afterAgreementCreated callbacks:
     AssertionError: Expected "0" to be equal 1000000000000000
      at Context.<anonymous> (test/REXUniswapTwoWayMarket.test.ts:663:18)
      at runMicrotasks (<anonymous>)
      at processTicksAndRejections (internal/process/task_queues.js:95:5)

  4) REXTwoWayUniswapMarket
       #2 - existing market with streamers on both sides
         #2.2 before/afterAgreementUpdated callbacks:
     Error: a provider or signer is needed to resolve ENS names (operation="resolveName", code=UNSUPPORTED_OPERATION, version=contracts/5.7.0)
      at Logger.makeError (node_modules/@ethersproject/logger/src.ts/index.ts:269:28)
      at Logger.throwError (node_modules/@ethersproject/logger/src.ts/index.ts:281:20)
      at /Users/mikeghen/Documents/Projects/test/ricochet-protocol/node_modules/@ethersproject/contracts/src.ts/index.ts:132:16
      at step (node_modules/@ethersproject/contracts/lib/index.js:48:23)
      at Object.next (node_modules/@ethersproject/contracts/lib/index.js:29:53)
      at fulfilled (node_modules/@ethersproject/contracts/lib/index.js:20:58)
      at runMicrotasks (<anonymous>)
      at processTicksAndRejections (internal/process/task_queues.js:95:5)

  5) REXTwoWayUniswapMarket
       #2 - existing market with streamers on both sides
         #2.3 two-sided distribution:
     Error: Transaction reverted: function returned an unexpected amount of data
      at REXTwoWayUniswapMarket._swap (contracts/REXUniswapTwoWayMarket.sol:402)
      at REXTwoWayUniswapMarket.distribute (contracts/REXUniswapTwoWayMarket.sol:196)
      at runMicrotasks (<anonymous>)
      at processTicksAndRejections (internal/process/task_queues.js:95:5)
      at runNextTicks (internal/process/task_queues.js:64:3)
      at listOnTimeout (internal/timers.js:526:9)
      at processTimers (internal/timers.js:500:7)
      at async HardhatNode._mineBlockWithPendingTxs (node_modules/hardhat/src/internal/hardhat-network/provider/node.ts:1649:23)
      at async HardhatNode.mineBlock (node_modules/hardhat/src/internal/hardhat-network/provider/node.ts:458:16)
```

**Comment by shreyaspapi** [comment]
Created a new pull request - https://github.com/Ricochet-Exchange/ricochet-protocol/pull/138

**Pull request closed without merging**